client/src/components/Input/index.js

```diff
@@ -4,24 +4,14 @@ import classnames from "classnames";
 
 import "./input.css";
 
-const Input = ({
-  labelText,
-  placeholder,
-  onChange,
-  className,
-  name,
-  type,
-  value
-}) => (
+const Input = ({ labelText, className, labelClassName, ...allAttributes }) => (
   <div className="container-input">
-    <label className="label-style">{labelText}</label>
+    <label className={classnames("label-style", labelClassName)}>
+      {labelText}
+    </label>
     <input
-      name={name}
       className={classnames("input-style", className)}
-      placeholder={placeholder}
-      onChange={onChange}
-      value={value}
-      type={type}
+      {...allAttributes}
```

Comment:
Again, this is your call, but on this line, you're giving dynamic attributes to a child with a parent which has static properties. For example, if I called your component as follows:
```jsx
<Input style={{ width: 100, height: 400 }} />
```
This would give my `style` attribute to your `input` changing it's height and width to exceed it's parent. Either change the parent's `CSS` or hand your `allAttributes` to your parent `div` instead of your `input`.

Reply:
I agree, and I'd say the same goes for how you're using `labelClassname` and `className`.  It might be simpler to put className on the top-level `.container-input` div and then use that classname as a modifier for all of the child elements, taking advantage of CSS's cascade (then you would find that you no longer need labelClassName, and can just rely on the className you've added `.container-input`).

For example, if you passed 'error' as a classname you could use that to make the label text red, the input have a red border, and so on.